### PR TITLE
Remove unused parameter timeoutMilliseconds from ProcessMetrics

### PR DESCRIPTION
**Pull request by utpilla**
## Changes
- Remove unused parameter `timeoutMilliseconds` parameter from `ProcessMetrics`

### CODE REVIEW

File: src/OpenTelemetry/Metrics/BaseExportingMetricReader.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ internal override void SetParentProvider(BaseProvider parentProvider)
         }
 
         /// <inheritdoc/>
-        internal override bool ProcessMetrics(in Batch<Metric> metrics, int timeoutMilliseconds)
+        internal override bool ProcessMetrics(in Batch<Metric> metrics)
```
Comment (reyang) [inline src/OpenTelemetry/Metrics/BaseExportingMetricReader.cs:81]: Why do we want to remove it?

Comment (utpilla) [reply src/OpenTelemetry/Metrics/BaseExportingMetricReader.cs:81]: Just to simplify the method signature because `ProcessMetrics` does not use it.

Comment (mic-max) [reply src/OpenTelemetry/Metrics/BaseExportingMetricReader.cs:81]: How did you find this?
Is there some stylecop rule that can help us clean up any other similar unused parameters for our private and internal functions?

Comment (utpilla) [reply src/OpenTelemetry/Metrics/BaseExportingMetricReader.cs:81]: I found this while working on a PR that involved adding logs to Metrics SDK code. There is a code quality rule for this: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0060

However, we don't want to remove this as we would want to modify `BaseExporter.Export` method to add this parameter.

**Comment by utpilla** [comment]
Spoke to @reyang offline. We intend to update the Export method signature in future to support timeout. So, we don't want to remove this.

**Pull request closed without merging**